FastPriorityQueue.js

```diff
@@ -124,40 +126,73 @@ FastPriorityQueue.prototype._percolateDown = function(i) {
 };
 
 // internal
-// _removeAt(index) will delete the given index from the queue,
-// retaining balance. returns true if removed.
+// _removeAt(index) will remove the item at the given index from the queue,
+// retaining balance. returns the removed item, or undefined if nothing is removed.
 FastPriorityQueue.prototype._removeAt = function(index) {
-  if (this.isEmpty() || index > this.size - 1 || index < 0) return false;
+  if (index > this.size - 1 || index < 0) return undefined;
 
   // impl1:
   //this.array.splice(index, 1);
   //this.heapify(this.array);
   // impl2:
   this._percolateUp(index, true);
-  this.poll();
-  return true;
+  return this.poll();
 };
 
-// remove(myval[, comparator]) will remove the given item from the
-// queue, checked for equality by using compare if a new comparator isn't provided.
-// (for exmaple, if you want to remove based on a seperate key value, not necessarily priority).
+// remove(myval) will remove the given item from the
+// queue, checked for equality by using compare.
 // return true if removed.
-FastPriorityQueue.prototype.remove = function(myval, comparator) {
-  if (!comparator) {
-    comparator = this.compare;
-  }
-  if (this.isEmpty()) return false;
+FastPriorityQueue.prototype.remove = function(myval) {
   for (var i = 0; i < this.size; i++) {
-    if (comparator(this.array[i], myval) || comparator(myval, this.array[i])) {
+    if (this.compare(this.array[i], myval) || this.compare(myval, this.array[i])) {
       continue;
     }
     // items are equal, remove
-    return this._removeAt(i);
+    return this._removeAt(i) !== undefined;
```

Comment:
Given than i is between 0 and this.size, why would this._removeAt(i) ever return 'undefined'? It shouldn't unless we have a bug of some sort.

Reply:
You're right, something would have to be very wrong for this to be undefined. I think I had this undefined check in here largely to return a boolean value, without really considering that it *should* always return true if it gets this far. I think it might be OK to leave this statement  as is though, so it returns `false` and informs the caller of the method if it's not removed for some reason. Another option is to throw an error if it's `undefined` at this point, but I noticed that errors are not being thrown anywhere else in this file. Let me know what you think is best.